releaseNotes/Microsoft.Featuremanagement.md

```diff
@@ -4,6 +4,90 @@
 # Microsoft.FeatureManagement.AspNetCore
 [Source code ][source_code_web] | [Package (NuGet)][package_web] | [Samples][samples_web] | [Product documentation][docs]
 
+## 4.0.0-preview2 - March 7, 2024
+
+### Feature Service Injector
+
+Dependency injection can be wired up with a variant feature flag. Different implementations of a service interface can be considered as variant services. Using the newly provided `IVariantServiceProvider<TService>`, the appropriate service can be injected.
+
+To use it, a variant service needs to be declared
+
+``` C#
+services.AddFeatureManagement() 
+        .WithVariantService<IAlgorithm>("ForecastAlgorithm");
+```
+
+The different options of the service need to extend a shared interface
+
+``` C#
+public class AlgorithmAlpha : IAlgorithm
+{
+    ...
+}
+
+public class AlgorithmBeta : IAlgorithm
+{
+    ...
+}
+```
+
+And the appropriate variant flag needs to exist
+
+```json
+    "ForecastAlgorithm": {
+        "Variants": [ 
+            { 
+                "Name": "AlgorithmAlpha" 
+            },
+            { 
+                "Name": "AlgorithmBeta" 
+            }
+        ] 
+    }
+```
+
+Lastly, the provider is ready to be used and grab the appropriate service for the variant.
+
+``` C#
+IVariantServiceProvider<IAlgorithm> algorithmServiceProvider;
+...
+IAlgorithm forecastAlgorithm = await algorithmServiceProvider.GetServiceAsync(cancellationToken);
+```
+
+The readme will be updated to fully describe the new functionality.
+
+### Telemetry & Targeting
+
+This preview release includes some additional mechanisms to track targeting within telemetry. There's been some changes to the telemetry publisher fields, and a couple classes are now offered/have been updated: `TargetingTelemetryInitializer` in `Microsoft.FeatureManagement.Telemetry.AspNetCore` and `TargetingHttpContextMiddleware` in `Microsoft.FeatureManagement.Telemetry.AspNetCore`. 
```

Comment:
Seems `TargetingTelemetryInitializer` and `TargetingHttpContextMiddleware` need to be mentioned in our readme.

Reply:
Which has been added, which has been updated?

Suggest doing like

* Class (added)
* Class2 (updated)